### ISSUE TITLE
Should we assume CEF3?

### ISSUE THREAD

**Issue by benjaminkay93** [NaSTAuk/graphics#13]
It appears that CasparCG supports at least CEF3239 Chromium 63 (see https://github.com/CasparCG/server/pull/724) as of 29 December 2018.

Should we assume going forward that we won't support less than this? I can imagine a world where graphics packages won't necessarily support something so up to date, but we could end up in a world where we have very limited CSS options if we don't draw a line modern enough.

**Comment by niklasR** [comment]
Has this been tested/confirmed by someone here? I'd like to be able to say yes, CEF3 with 3d transforms and whatnot 👍, but it would be nice to have it tested by someone in Caspar, rendered out on a decklink or so.

Could use a suite like http://tests.caniuse.com/ to define what a stable Caspar 2.2 supports out the box and fix to that?

I do not have the capability to do that.

**Comment by benjaminkay93** [comment]
That's a cool link, some tooling to valid our usage would be great to have too! Chromium 63 doesnt support gyroscopes though 😞 
![image](https://user-images.githubusercontent.com/10714808/56302234-41fe7e00-6131-11e9-80d7-7053f052e95f.png)

It would nice to have some internal validation for CEF 3 before locking into it. I don't have the necessary things either, @thomasslee97 or @m3lnic, any chance of testing the CEF Version on CasparCG at either of your stations? I can like to an HTML file that would make it clear if it works ;P (@moschopsuk you get one guess)

**Comment by benjaminkay93** [comment]
https://github.com/LA1TV/Roses-CasparCG-Graphics/blob/master/public/perspective/field_view.html

Is a file to run to see if CEF 3 works 😄

**Comment by Vitupero** [comment]
I would assume that this works if it's been merged, but given that Julian (Ex-Guild TV) wrote the PR, I'll ask him!

**Comment by benjaminkay93** [comment]
Yeah, we should be relatively safe. It's probably best to just validate our assumptions, even if that is going to the person who implemented it :')

**Comment by thomasslee97** [comment]
Yup, current stable CasparCG is CEF3 and supports perspective :)

![image](https://user-images.githubusercontent.com/10697525/56329208-5f0b6f00-617a-11e9-86ee-c9333fd1ac52.png)

**Comment by thomasslee97** [comment]
Result of gyroscope test: negative
![image](https://user-images.githubusercontent.com/10697525/56329414-49e31000-617b-11e9-845c-cc20223d8600.png)

**Comment by benjaminkay93** [comment]
Well, that looks beautiful :O 

Such a shame about the Gyro though, imagine all the possibilities, we could have known the orientation of a server in any rack... spoiler alert, it is probably horizontal, but it would be nice to have that data!

**Comment by benjaminkay93** [comment]
Thanks for that @thomasslee97 😄 🚀

**Comment by moschopsuk** [comment]
> Yup, current stable CasparCG is CEF3 :)
> 
> ![image](https://user-images.githubusercontent.com/10697525/56329208-5f0b6f00-617a-11e9-86ee-c9333fd1ac52.png)

Looks like some of the textures are not rendering correctly ?

**Comment by benjaminkay93** [comment]
maybe the image texture doesn't exsist anymore?

**Comment by thomasslee97** [comment]
This looks to be the missing texture, I'll give it another go at some point + see if there's something obvious I've done wrong.

http://i.imgur.com/bh9ahq3.png

**Comment by thomasslee97** [comment]
Yeah the playing field texture wasn't loading from Imgur (tested on Chrome and Firefox). I've stuck it in the /images directory and loaded it from there. Works in CasparCG now!

![image](https://user-images.githubusercontent.com/10697525/57587673-931b4b00-7500-11e9-8958-7ec0fc8453fb.png)

**Comment by moschopsuk** [comment]
Nice work @thomasslee!
